[Bug]: Unable to retrieve triggered stop order using gRPC method GetStopOrder

### Problem encountered

Unable to retrieve triggered stop order using gRPC method GetStopOrder

Note: issue occurs when retrieving stop orders with status TRIGGERED
PENDING stop orders seem to be fine.

### Observed behaviour

Receive an error when retrieving a triggered stop order using the gRPC method GetStopOrder

```
>           raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
E           grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
E           	status = StatusCode.INTERNAL
E           	details = "Internal error"
E           	debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:3027 {created_time:"2023-07-10T16:32:36.082943+01:00", grpc_status:13, grpc_message:"Internal error"}"
E           >
```

Similar error for graphQL and REST end points as well.

```
query OneStopOrder {
  stopOrder (id: "b47fb4662dc1cbb8d5dc3f0208358eca1b1f7ffe0232501e279893d488c23186") {
    id
    ocoLinkId
    status
  }
```

Response

```
{
  "errors": [
    {
      "message": "order not found",
      "path": [
        "stopOrder"
      ],
      "extensions": {
        "code": 13,
        "type": "Internal"
      }
    }
  ],
  "data": {
    "stopOrder": null
  }
}
```

### Expected behaviour

Should be able to retrieve stop orders successfully.

### Steps to reproduce

```bash
Run the below system test :

System Test Name - test_stop_order_once_triggered_can_not_be_cancelled
System Test Branch - feat_stop_orders_2
```


### Software version

vega - develop

### Failing test

test_stop_order_once_triggered_can_not_be_cancelled

### Jenkins run

https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/82407/testReport/junit/tests.orders/stop_orders_test/test_stop_order_once_triggered_can_not_be_cancelled_order_param0_/

### Configuration used

_No response_

### Relevant log output

```shell
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INTERNAL
	details = "Internal error"
	debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:3027 {created_time:"2023-07-10T15:52:49.363678536+00:00", grpc_status:13, grpc_message:"Internal error"}"
>
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -32,6 +32,7 @@
 - [8713](https://github.com/vegaprotocol/vega/issues/8713) - Wallet name with upper-case letter is allowed, again
 - [8698](https://github.com/vegaprotocol/vega/issues/8698) - Always set the `ToAccount` field when clearing fees.
 - [8711](https://github.com/vegaprotocol/vega/issues/8711) - Fix market lineage trigger.
+- [8737](https://github.com/vegaprotocol/vega/issues/8737) - Fix `GetStopOrder` errors when order has a state change.
 
 
 ## 0.72.0

--- a/datanode/sqlstore/stop_orders.go
+++ b/datanode/sqlstore/stop_orders.go
@@ -66,7 +66,7 @@ func (so *StopOrders) GetStopOrder(ctx context.Context, orderID string) (entitie
 	order := entities.StopOrder{}
 	id := entities.StopOrderID(orderID)
 	defer metrics.StartSQLQuery("StopOrders", "GetStopOrder")()
-	query := `select * from stop_orders where id=$1`
+	query := `select * from stop_orders_current_desc where id=$1`
 	err = pgxscan.Get(ctx, so.Connection, &order, query, id)
 
 	return order, so.wrapE(err)

--- a/datanode/sqlstore/stop_orders_test.go
+++ b/datanode/sqlstore/stop_orders_test.go
@@ -145,26 +145,40 @@ func TestStopOrders_Get(t *testing.T) {
 	defer rollback()
 
 	block := addTestBlock(t, ctx, bs)
+	block2 := addTestBlock(t, ctx, bs)
 
 	party := addTestParty(t, ctx, ps, block)
 
 	markets := helpers.GenerateMarkets(t, ctx, 1, block, ms)
 
+	orderID := helpers.GenerateID()
 	stopOrders := generateTestStopOrders(t, []testStopOrderInputs{
 		{
-			orderID:        helpers.GenerateID(),
+			orderID:        orderID,
 			vegaTime:       block.VegaTime,
 			createdAt:      block.VegaTime,
 			triggerPrice:   "100",
 			partyID:        party.ID,
 			marketID:       markets[0].ID,
-			status:         entities.StopOrderStatusUnspecified,
+			status:         entities.StopOrderStatusPending,
+			expiryStrategy: entities.StopOrderExpiryStrategyUnspecified,
+		},
+		{
+			orderID:        orderID,
+			vegaTime:       block2.VegaTime,
+			createdAt:      block.VegaTime,
+			triggerPrice:   "100",
+			partyID:        party.ID,
+			marketID:       markets[0].ID,
+			status:         entities.StopOrderStatusTriggered,
 			expiryStrategy: entities.StopOrderExpiryStrategyUnspecified,
 		},
 	})
 
-	err := so.Add(stopOrders[0])
-	require.NoError(t, err)
+	for i := range stopOrders {
+		err := so.Add(stopOrders[i])
+		require.NoError(t, err)
+	}
 
 	want, err := so.Flush(ctx)
 	require.NoError(t, err)
@@ -176,9 +190,9 @@ func TestStopOrders_Get(t *testing.T) {
 	})
 
 	t.Run("Get should return the order if the order ID does exist", func(t *testing.T) {
-		got, err := so.GetStopOrder(ctx, stopOrders[0].ID.String())
+		got, err := so.GetStopOrder(ctx, orderID)
 		require.NoError(t, err)
-		assert.Equal(t, want[0], got)
+		assert.Equal(t, want[1], got)
 	})
 }
 

--- a/go.mod
+++ b/go.mod
@@ -47,6 +47,7 @@ require (
 	github.com/adrg/xdg v0.4.0
 	github.com/blang/semver/v4 v4.0.0
 	github.com/cenkalti/backoff/v4 v4.2.0
+	github.com/cometbft/cometbft-db v0.7.0
 	github.com/cucumber/messages-go/v16 v16.0.1
 	github.com/dgraph-io/badger/v2 v2.2007.4
 	github.com/dgrijalva/jwt-go/v4 v4.0.0-preview1
@@ -112,7 +113,6 @@ require (
 	github.com/cockroachdb/logtags v0.0.0-20230118201751-21c54148d20b // indirect
 	github.com/cockroachdb/pebble v0.0.0-20230209160836-829675f94811 // indirect
 	github.com/cockroachdb/redact v1.1.3 // indirect
-	github.com/cometbft/cometbft-db v0.7.0 // indirect
 	github.com/containerd/cgroups v1.1.0 // indirect
 	github.com/coreos/go-systemd/v22 v22.5.0 // indirect
 	github.com/cosmos/go-bip39 v0.0.0-20180819234021-555e2067c45d // indirect

